### PR TITLE
[API][Web][SIMS #1466] Remove date only transformer for date only columns

### PR DESCRIPTION
**Pull request by dheepak-aot**
## Remove date only transformer for all date only columns in entity model

- [x] Remove dateOnlyTransformer from all the date-only properties in entity model and change the data type to string.
- [x] Ensure that fields mapped as date-only on DB have the type declared as `type: "date"` in the entity model, as per the screenshot above.
- [x] Remove the `getISODateOnlyString` conversion as needed usually used when the API is converting the data-time to a date-only string to be sent to the API.
- [x] Test the student application data to have selectedOfferingDate as date-only string.
image

There are one scenario where the date fields are read as date objects and converted to ISO date only string and vice-versa.
- File integrations

### File integration conversions

During the file integration process all the date only fields obtained by reading a file or while creating a file is not in ISO date only format. For this reason, the date fields when read from file(s) during the integration are converted to date object and then converted to ISO date only string to be saved in database.

### CODE REVIEW

File: sources/packages/backend/apps/api/src/esdc-integration/msfaa-integration/msfaa-response.service.ts [hunk 1]
```diff
@@ -102,10 +103,13 @@ export class MSFAAResponseService {
...
     receivedRecord: MSFAAResponseReceivedRecord,
   ): Promise<void> {
+    // The update of msfaa always comes from an external source through integration.
+    // Hence the date fields are parsed as date object from external source as their date format
+    // may not be necessarily ISO date format.
     const updateResult = await this.msfaaNumberService.updateReceivedFile(
       receivedRecord.msfaaNumber,
-      receivedRecord.borrowerSignedDate,
-      receivedRecord.serviceProviderReceivedDate,
+      getISODateOnlyString(receivedRecord.borrowerSignedDate),
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/esdc-integration/msfaa-integration/msfaa-response.service.ts:111]: For these situations would not be better for the `updateReceivedFile` to receive the Date type object and then do the conversion internally?

File: sources/packages/backend/apps/api/src/route-controllers/application/models/application.model.ts [hunk 2]
```diff
@@ -91,8 +91,8 @@ export class GetApplicationDataDto extends GetApplicationBaseDTO {
   applicationAssessmentStatus?: AssessmentStatus;
   applicationPIRDeniedReason?: string;
   applicationCOEDeniedReason?: string;
-  programYearStartDate: Date;
-  programYearEndDate: Date;
+  programYearStartDate: string;
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/models/application.model.ts:94]: I added this file to the tech debt: https://app.zenhub.com/workspaces/student-information-management-system-5fce9df5aa1b45000e937014/issues/bcgov/sims/1495

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.institutions.controller.ts [hunk 3]
```diff
@@ -193,9 +192,7 @@ export class ConfirmationOfEnrollmentInstitutionsController extends BaseControll
       })),
       applicationPIRStatus:
         disbursementSchedule.studentAssessment.application.pirStatus,
-      disbursementDate: getExtendedDateFormat(
-        disbursementSchedule.disbursementDate,
-      ),
+      disbursementDate: disbursementSchedule.disbursementDate,
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.institutions.controller.ts:195]: is this `getExtendedDateFormat` taken care of by the vue side?

File: sources/packages/backend/apps/api/src/route-controllers/education-program/education-program.aest.controller.ts [hunk 4]
```diff
@@ -94,7 +94,7 @@ export class EducationProgramAESTController extends BaseController {
     @Body() payload: ApproveProgramAPIInDTO,
   ): Promise<void> {
     await this.programService.approveEducationProgram(
-      new Date(payload.effectiveEndDate),
+      payload.effectiveEndDate,
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/education-program/education-program.aest.controller.ts:97]: shouldn't this be a a date object 
image

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.institutions.controller.ts (hunk 3 above)
Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.institutions.controller.ts:195]: Yes, it is converted at Vue side.

File: sources/packages/backend/apps/api/src/services/student/student.service.ts [hunk 5]
```diff
@@ -163,7 +164,7 @@ export class StudentService extends RecordDataModelService<Student> {
     }
 
     student.user = user;
-    student.birthDate = new Date(userInfo.birthdate);
+    student.birthDate = getISODateOnlyString(new Date(userInfo.birthdate));
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/services/student/student.service.ts:167]: The `getISODateOnlyString` accepts either a Date or string, conversions are not needed.

File: sources/packages/backend/apps/api/src/route-controllers/education-program/education-program.aest.controller.ts (hunk 4 above)
Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/route-controllers/education-program/education-program.aest.controller.ts:97]: effectiveEndDate is a string in entity model as it is date only column now.

File: sources/packages/backend/libs/sims-db/src/entities/msfaa-number.model.ts [hunk 6]
```diff
@@ -82,7 +79,7 @@ export class MSFAANumber extends RecordDataModel {
     name: "cancelled_date",
     nullable: true,
   })
-  cancelledDate?: Date;
+  cancelledDate?: string;
```
Comment (andrewsignori-aot) [inline sources/packages/backend/libs/sims-db/src/entities/msfaa-number.model.ts:82]: Missing `type: "date"`.

File: sources/packages/backend/apps/api/src/services/student/student.service.ts (hunk 5 above)
Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/services/student/student.service.ts:167]: I believe I can directly update like this without any conversion `student.birthDate = userInfo.birthdate;` as `userInfo.birthdate` is the token birthdate value.

File: sources/packages/web/src/services/http/dto/Institution.dto.ts [hunk 7]
```diff
@@ -44,6 +44,7 @@ export interface InstitutionProfileAPIInDTO extends InstitutionContactAPIInDTO {
 export interface InstitutionDetailAPIOutDTO
   extends InstitutionProfileAPIOutDTO {
   legalOperatingName: string;
+  // TODO: Need to be removed. Not removed here as it fails to compile the cypress code.
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/services/http/dto/Institution.dto.ts:47]: Can we please mark those ones with `@deprecated`?

**Review by andrewsignori-aot — COMMENTED** [review]
Great work 😉, just minor comments, please take a look.

File: sources/packages/backend/apps/api/src/route-controllers/education-program/education-program.aest.controller.ts (hunk 4 above)
Comment (ann-aot) [reply sources/packages/backend/apps/api/src/route-controllers/education-program/education-program.aest.controller.ts:97]: pls, ignore the comment. i saw the change

**Review by ann-aot — APPROVED** [review]
Great work @dheepak-aot 👍 Thanks for doing the changes

File: sources/packages/backend/apps/api/src/esdc-integration/msfaa-integration/msfaa-response.service.ts (hunk 1 above)
Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/esdc-integration/msfaa-integration/msfaa-response.service.ts]: Changes done as we discussed.

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes. Looks good 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @dheepak-aot

**Comment by andrepestana-aot** [comment]
LGTM. There's just a code smell.

**Pull request merged**